src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -513,6 +545,28 @@ private void AddCentralTransitiveDependencyGroupsForPackageReference(PackageSpec
             }
         }
 
+        /// <summary>
+        /// Returns a flattened list of all parent nodes for the specified node.
+        /// </summary>
+        /// <typeparam name="T">The type of the node.</typeparam>
+        /// <param name="graphNode">The <see cref="GraphNode{TItem}" /> to flatten the parent nodes of.</param>
+        /// <returns>A top down flattened list of parent nodes of the specied node.</returns>
+        private IEnumerable<GraphNode<T>> FlattenParentNodes<T>(GraphNode<T> graphNode)
```

Comment:
Can this method be static since there's no modifications on instance members?
```suggestion
        private static IEnumerable<GraphNode<T>> FlattenParentNodes<T>(GraphNode<T> graphNode)
```